frontend/src/routes/login.lazy.tsx

```diff
@@ -65,18 +66,94 @@ function Login() {
             Log in
           </h1>
           <div className="card flex flex-col gap-3">
-            {data?.errors?.root && (
-              <div style={{ color: "red" }}>{data.errors.root}</div>
-            )}
-            <div>
+            <Form.Root>
+              {data?.errors && (
+                <Form.Message
+                  className="text-red-500 text-sm"
+                  match={"valueMissing"}
+                >
+                  {data.errors.root}
+                </Form.Message>
+              )}
+
+              <Form.Field className="my-5" name="username">
+                {!!data?.errors?.username && (
+                  <Form.Message
+                    className="text-red-500 text-sm"
+                    match={"valueMissing"}
+                  >
+                    {data.errors.username}
+                  </Form.Message>
+                )}
+
+                <Form.Control asChild>
+                  <Input
+                    placeholder="username"
+                    aria-label="username"
+                    name="username"
+                    type="text"
+                  />
+                </Form.Control>
+              </Form.Field>
+
+              <Form.Field className="my-5" name="password">
```

Comment:
```suggestion
              <Form.Field className="my-2" name="password">
```

Reduce the spacing between fields